notes/hashfunctions3.tex

```diff
@@ -0,0 +1,353 @@
+\section{PRFs from Hash functions}
+
+\subsection{Other Hash based PRFs}
+The Merkle Damg{\aa}rd construction from the previous section is a type of Prefix MAC (since the key is prepended to the message). Although the construction is secure PRF in the random oracle model, we saw how it is susceptible to length extension attacks. Several alternatives have been proposed to counter these attacks. 
+
```

Comment:
Define what hash^f is in the following constructions? Namely, the Merkle-Damgard domain extending transform with a compression function f.